### PR TITLE
Fix Multiclass layer gradient

### PR DESCRIPTION
**Pull request by andrewdalpino**
Softmax activation function is incompatible with any non-Cross Entropy-based loss function. Remove Softmax activation function. Subsume functionality into Multiclass layer - correct the gradient computation for non-Cross Entropy-based loss function such as Relative Entropy.

### CODE REVIEW

**Review by apphp — APPROVED** [review]

**Comment by andrewdalpino** [comment]
Thank you @apphp!
